tftb/generators/misc.py

```diff
@@ -174,58 +185,72 @@ def mexhat(nu=0.05):
     return h
 
 
-def gdpower(n_points, degree=0, rate=1):
+def gdpower(n_points, degree=0.0, rate=1.0):
     """Generate a signal with a power law group delay.
 
     :param n_points: Number of points in time.
     :param degree: degree of the power law.
     :param rate: rate-coefficient of the power law GD.
     :type n_points: int
-    :type degree: int
+    :type degree: int		# yoder: i'm pretty sure this is supposed to be a float. at least in test_misc.py, the test for this funct. passes degree=0.5
     :type rate: float
     :return: Tuple of time row containing modulated samples, group delay, \
             frequency bins.
     :rtype: tuple
     """
+    # quickly, handle types:
+    n_points = int(n_points)
+    degree   = float(degree)
+    rate     = float(rate)
+    #print("diagnostic: running gdpower")
     t0 = 0
-    lnu = np.round(n_points / 2)
+    #lnu = int(np.round(n_points / 2))
+    lnu = int(np.ceil(n_points / 2))		# ??
     nu = np.linspace(0, 0.5, lnu + 1)
     nu = nu[1:]
-    am = nu ** ((degree - 2) / 6)
+    #am = nu ** ((degree - 2) / 6)
+    am = nu ** ((degree - 2.0) / 6.0)
 
-    if rate == 0:
+    if rate == 0.:
         raise TypeError("rate must be non-zero")
 
     tfx = np.zeros((n_points,), dtype=complex)
 
-    if (degree < 1) and (degree != 0):
-        d = n_points ** (degree * rate)
+    if (degree < 1.) and (degree != 0):
+        #d = n_points ** (degree * rate)
+        d = float(n_points) ** (degree * rate)
         t0 = n_points / 10.0
-        tfx[:lnu] = np.exp(-1j * 2 * pi * (t0 * nu + d * nu ** degree / degree)) * am
+        #tfx[:lnu] = np.exp(-1j * 2 * pi * (t0 * nu + d * nu ** degree / degree)) * am
+        tfx[:lnu] = np.exp(-1.0j * 2. * pi * ((t0 * nu + d * nu ** degree) / float(degree))) * am
```

Comment:
Ah! Here's the problem. This right side of this equation is:
```python
np.exp(-1j * 2 * pi * (t0 * nu + d * nu ** degree / degree) * am
```
and what's actually being computed is:
```python
np.exp(-1j * 2 * pi * ((t0 * nu + d * nu ** degree) / degree) * am
```
Also, no need for `float(degree)` since it's been casted in #L203